src/content/courses/basic-theory/chapter_5.mdx

```diff
@@ -8,26 +8,30 @@ Lock is a script structure that looks like this:
 Script: {
   code_hash: HexString
   args: HexString
-  hash_type: either 'data' or 'type'
+  hash_type: Uint8, there are three allowed values: {0: "data", 1: "type", 2: "data1"}
 }
 ```
 
-You may notice that the code_hash is not the actual code but a hash of the code, equivalent to an index of the code. This index allows us to retrieve the code. So, where is the code anyway?
+You may notice that the `code_hash` is not the actual code, but some kind of index of the code. This index allows us to retrieve the code. So, where is the code anyway?
 
-The answer is simple: code is in another cell!
+The answer is simple: the code is stored in another cell!
 
-We know that the data field of the cell can contain arbitrary data, so we can put the real code in the data field of another cell and implement this cell as a dependency to a transaction. This dependency cell is called “dep cell”.
+We know that the data field of a cell can contain arbitrary data, so we can put the real code in the data field of another cell and implement this cell as a dependency to a transaction. This dependency cell is called `CellDep` (dep cell).
 
-When unlocking a cell, we simply import the dep cell, and CKB will match the hash of the data in the dep cell with code_hash to find the corresponding code.
+Depending on the value of `hash_type`, `code_hash` has different interpretations:
+- If hash_type is "data" or "data1", code_hash should match the blake2b hash of data in a dep cell;
+- If hash_type is "type", code_hash should instead match the type script hash of a dep cell.
```

Comment:
can a newbie understand `the type script hash of a dep cell`?

Reply:
That's a question.
Maybe we should add some explanation for `the type script hash of a dep cell` or a link to this topic.

The example `SECP256K1_BLAKE160`'s hash_type is `type`. 
To avoid misunderstandings, I think it's better to explain this topic before showing the example.

any suggestion?

Reply:
yea it is worth explaining the concept or giving it a link.

Reply:
some process flow in graph to simple display the hashing process of hash_type

Reply:
It seems that all the common lock/type scripts are using `hash_type=type`.

see https://pudge.explorer.nervos.org/scripts